pg_conversion.c

```diff
@@ -334,78 +336,233 @@ pg_array_get_r(Datum dvalue, FmgrInfo out_func, int typlen, bool typbyval, char
+					continue;
+
+				/* set column names */
+				if (numels == 0)
+					SET_COLUMN_NAMES;
+
+				/* update column datatype oid and check for embedded array */
+				element_type = SPI_gettypeid(tupdesc, j + 1);
+				typelem = get_element_type(element_type);
+				if (typelem != InvalidOid)
+				{
+					get_type_io_data(typelem, IOFunc_output, &typlen, &typbyval, &typalign, &typdelim, &typioparam, &typoutput);
+					fmgr_info(typoutput, &outputproc);
+				}
+			}
+
+			if (numels == 0)
+			{
+				/* allocate new vector of the appropriate type and length */
+				if (typelem == InvalidOid)
+					/* dealing with scalars of element_type */
+					PROTECT(v = get_r_vector(element_type, nr));
+				else
+					/* dealing with arrays (containing typelem's) */
+					PROTECT(v = NEW_LIST(nr));
+				SET_VECTOR_ELT(result, df_colnum, v);
+				UNPROTECT(1);
+			}
+
+			v = VECTOR_ELT(result, df_colnum);
+			if (!isrel && typelem == InvalidOid)
+			{
+				/* scalar type */
+				char	   *value;
+				switch (element_type)
+				{
+					case BOOLOID:
+						LOGICAL_DATA(v)[numels] = isnull ? NA_LOGICAL : DatumGetBool(dvalue);
+						break;
+					case INT8OID:
+						NUMERIC_DATA(v)[numels] = isnull ? NA_REAL : (double)DatumGetInt64(dvalue);
+						break;
+					case INT2OID:
+					case INT4OID:
+					case OIDOID:
+						INTEGER_DATA(v)[numels] = isnull ? NA_INTEGER : DatumGetInt32(dvalue);
+						break;
+					case FLOAT4OID:
+						NUMERIC_DATA(v)[numels] = isnull ? NA_REAL : DatumGetFloat4(dvalue);
+						break;
+					case FLOAT8OID:
+						NUMERIC_DATA(v)[numels] = isnull ? NA_REAL : DatumGetFloat8(dvalue);
+						break;
+					default:
+						value = isnull ? NULL :
+							DatumGetCString(FunctionCall3(&out_func, dvalue, (Datum) 0, Int32GetDatum(-1)));
+						/*
+						 * Note that pg_get_one_r() replaces NULL values with
+						 * the NA value appropriate for the data type.
+						 */
+						pg_get_one_r(value, element_type, v, numels);
+						if (value != NULL)
+							pfree(value);
+				}
+			}
+			else if (isrel && typelem == InvalidOid)
+			{
+				char *value = isnull ? NULL : SPI_getvalue(tuple, tupdesc, j + 1);
+				pg_get_one_r(value, element_type, v, numels);
 				if (value != NULL)
 					pfree(value);
+			}
+			else /* typelem != InvalidOid, i.e.: */
+			{
+				/* array type (regardless of whether embedded in a tuple or not) */
+				SEXP		fldvec_elem;
+				Datum	   value = dvalue;
+				bool		isvaluenull = isnull;
+				if (isrel && !isnull)
+					value = SPI_getbinval(tuple, tupdesc, j + 1, &isvaluenull);
+
+				if (!isvaluenull)
+					PROTECT(fldvec_elem = pg_array_get_r(value, outputproc, typlen, typbyval, typalign));
+				else
+					PROTECT(fldvec_elem = R_NilValue);
+				SET_VECTOR_ELT(v, numels, fldvec_elem);
+				UNPROTECT(1);
+			}
+			df_colnum++;
 		}
+
+		if (isrel && !isnull)
+			pfree(tuple);
 	}
 
-	if (numels != totalrows)
-		SET_LENGTH(result, numels);
+	/* Resize all vectors from nr (rows in partition) down to numels (rows in frame) */
```

Comment:
if nr really means rows in partition why not use that name for the variable? Same with numels ?

Reply:
At the time I chose to stick with existing (historical) code conventions I guess

Reply:
Can you change the names please ? As far as I know the reason they used short names historically was because it was difficult to type long names. We no longer have that problem.

Reply:
I have changed the variable names but have made a mess of the commit history in the process. I will see if I can do a force push to cleanup the history again.